### PR TITLE
Move to merkledb

### PR DESCRIPTION
**Pull request by pavel-mukhanov**
## Overview
Attempt to move to MerkledDB with all history.

---

### Definition of Done

- [x] There are no TODOs left in the merged code
- [x] Change is covered by automated tests
- [x] Benchmark results are attached (if applicable)
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: components/merkledb/.gitignore [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# vim files
```
Comment (alekseysidorov) [inline components/merkledb/.gitignore]: So I think we can just remove this file and above.

File: components/merkledb/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+version = "0.10.0"
+edition = "2018"
+authors = ["The Exonum Team <exonum@bitfury.com>"]
+homepage = "https://exonum.com/"
+repository = "https://github.com/exonum/exonum"
+documentation = "https://docs.rs/exonum-merkledb"
+readme = "README.md"
+license = "Apache-2.0"
+
+# keywords = ["exonum", "cryptography", "library"]
```
Comment (alekseysidorov) [inline components/merkledb/Cargo.toml]: I think later we should prepare this crate for publishing

File: examples/cryptocurrency-advanced/backend/src/schema.rs [hunk 3]
```diff
@@ -59,23 +58,20 @@ where
...
     /// Returns the state hash of cryptocurrency service.
     pub fn state_hash(&self) -> Vec<Hash> {
-        vec![self.wallets().merkle_root()]
+        vec![self.wallets().object_hash()]
     }
 }
 
 /// Implementation of mutable methods.
-impl<'a> Schema<&'a mut Fork> {
+impl<'a> Schema<&'a Fork> {
```
Comment (alekseysidorov) [inline examples/cryptocurrency-advanced/backend/src/schema.rs]: We can remove such specialisations.

**Comment by aleksuss** [comment]
@pavel-mukhanov resolve conflicts please

File: components/merkledb/.travis.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
+language: rust
```
Comment (aleksuss) [inline components/merkledb/.travis.yml]: @pavel-mukhanov This is a redundant file, isn't it ?

File: components/merkledb/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# Changelog
```
Comment (aleksuss) [inline components/merkledb/CHANGELOG.md]: @pavel-mukhanov I suppose this content should be moved to the main CHANGELOG file ?

**Comment by aleksuss** [comment]
@pavel-mukhanov resolve conflicts please.

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,10 +9,56 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
 #### exonum
 
+- Changed storage layout (#1293)
```
Comment (aleksuss) [inline CHANGELOG.md]: @pavel-mukhanov I suppose this related block to MerkleDB should be moved under **exonum-merkledb** section

**Pull request merged**